### ISSUE TITLE
URI::NamespaceMap welcomes patches 

### ISSUE THREAD

**Issue by kjetilk** [sipwise/data-hal#2]
Hi!

I'm the maintainer of URI::NamespaceMap, and I've had an issue open for Data::HAL breakage open for some time: https://github.com/kjetilk/URI-NamespaceMap/issues/9 

I gained some more understanding of the type system recently, so I found I could add coercions for RDF::Trine and Attean without depending on them, and I would certainly be open to do so for Data::HAL too.

I looked into it, and made a patch, https://github.com/kjetilk/URI-NamespaceMap/pull/17 but noticed it doesn't necessarily do much to help.

Another option is to make a coercion for URI::Template. That is possibly more attractive, since that implements an RFC. I would be fine with adding a coercion from URI::Template even if that required a dep on URI::Template. 

Anyway, bottom line is, I think we should be able to accomodate for your needs within URI::NamespaceMap, lets see if we can do this.

**Comment by taurus-forever** [comment]
Thank you for taking care here and provided us your findings.
The Data::HAL component was really developed in Sipwise and published to OSS here.
You can find the latest version we have here on GitHub.

Unfortunately author and CPAN maintainer has left the company long time ago and the component had not enough of love. Your report has been detected and will be processed internally, unfortunately with low priority. The Data::HAL did the job for Sipwise well, while for sure we would like to share those benefits with other OSS members. Your pull requests are highly appreciated here.

Thank you!

**Comment by kjetilk** [comment]
You're welcome! Lets see what happens, if someone gets around to check it out.
